great_expectations/data_context/store/datasource_store.py

```diff
@@ -113,12 +121,18 @@ def deserialize(
             return value
         elif isinstance(value, dict):
             # presence of a 'type' field means it's a fluent datasource
-            type_ = value.get("type")
+            type_: str | None = value.get("type")
             if type_:
-                datasource_model = _SourceFactories.type_lookup.get(type_)
-                if not datasource_model:
-                    raise LookupError(f"Unknown Datasource 'type': '{type_}'")
-                return datasource_model(**value)
+                try:
+                    datasource_model = _SourceFactories.type_lookup[type_]
+                    return datasource_model(**value)
+                except (PydanticValidationError, LookupError) as config_error:
+                    warnings.warn(
```

Comment:
@billdirks Since you last reviewed, I added handling for an error due to a `TypeLookup` error. When we talked last, I said I would do this as follow-up work, but it ended up being simpler than I expected.

The 2 scenarios that this covers are.
a. The datasource has a `type` that is not recognized.
b. An asset for a dynamically generated Datasource (pandas) has a type that is not recognized.